### ISSUE TITLE
No good error handling when GraphicsMagick or ImageMagick aren't installed

### ISSUE THREAD

**Issue by samholmes** [aheckmann/gm#273]
I've racked my brain over an error for TWO WHOLE DAYS, simply because of the fact that gm doesn't report an error when the proper libs (GraphicsMagick/ImageMagick) aren't installed on the system.

Frankly, I'm a little pissed. Had gm reported something like "Error: Missing GraphicsMagick or ImageMagick." when I attempt to use it, I wouldn't have spent all this time trying to figure out what's wrong with my photo uploading script. I thought, that it was AWS S3 storage (knox module) that was causing problems this whole time. After spending many hours debugging my application in production I was able to figure out that my app was crashing with these errors:

```
Error: read ECONNRESET
  at errnoException (net.js:904:11)
  at Pipe.onread (net.js:558:19)

Error: spawn ENOENT
  at errnoException (child_process.js:988:11)
  at Process.ChildProcess._handle.onexit (child_process.js:779:34)
```

all because the binaries weren't installed.

So, please, for the sake of other's who encounter similar situations, error out with an error explaining that GraphicsMagick or ImageMagick binaries aren't installed. This would be so useful. Thank you, thank you.

**Comment by Ferrari** [comment]
Any update on this issue?
Waste a long time for debug and figure out the root cause eventually, 
might be better show warning at the beginning.

**Comment by bobjones** [comment]
+1 huge pain.

**Comment by cport1** [comment]
FML

**Comment by rwky** [comment]
Fixed in dcefd670630e259e93a23ca592bf16b1fee6fc0f
